packages/hardhat-core/src/internal/artifacts.ts

```diff
@@ -78,7 +78,18 @@ export class Artifacts implements IArtifacts {
   }
 
   public async artifactExists(name: string): Promise<boolean> {
-    const artifactPath = await this._getArtifactPath(name);
+    let artifactPath = "";
+    try {
+      artifactPath = await this._getArtifactPath(name);
+    } catch (e) {
+      if (e instanceof HardhatError) {
```

Comment:
In general we use:

```suggestion
      if (HardhatError.isHardhatError(e)) {
```

To be honest, I always forget in which cases `instanceof` doesn't work. @alcuadrado do you remember this?